transcendence-app/src/user/infrastructure/db/user.entity.ts

```diff
@@ -0,0 +1,19 @@
+import { BaseEntity } from 'src/shared/db/models';
+
+export enum userKeys {
+  ID = 'id',
+  USERNAME = 'username',
+  EMAIL = 'email',
+  AVATAR = 'avatar',
+  TIMESTAMP = 'timestamp',
+}
+
+export class UserEntity implements BaseEntity {
+  constructor(
+    public username: string,
+    public email: string,
+    public avatar: string | null,
+    public id: string,
+    public created_at: number,
```

Comment:
why is this a number rather than a Date, for instance?
I guess all the fields that come from UserDto don't need validation, since they are validated there, but what about the one that is not?
Is @IsDate() not going to work for some reason?

Reply:
I'm gonna read carefully the conversation below on the three userXXX we have. User, UserEntity, UserDto. I'm not sure why we don't validate outside of Dto.

Reply:
Also we should agree on the case of our names.

CamelCase? pascalCase? snake_case? SCREAMING_CAMEL_CASE?

We have a mess.

In my understanding it should be CamelCase for exported symbols, pascalCase for local vars. We don't use snake_case.

Reply:
> Is @IsDate() not going to work for some reason?

The `created_at` entry is of time `TIMESTAMPZ` which is indeed utc time (a number).

Also, we are not suposed to validate anything here, that should be done in the `User` itself

Reply:
> Also we should agree on the case of our names.
> 
> PascalCase? camelCase? snake_case? SCREAMING_SNAKE_CASE?
> 
> We have a mess.
> 
> In my understanding it should be PascalCase for exported symbols, camelCase for non exported symbols (but we don't respect this strictly). We don't use snake_case. And we have one curious and funny instance of SCREAMING_SNAKE_CASE, that I don't particularly object to since I'm very much a C person

Agree, for me it would be:

- PascalCase for class names, interfaces etc.
- CamelCase for class instances (objects), variables, params etc
- SCREAMING_SNAKE_CASE for constants and env variables

I'd say this is a pretty standard way to go